### DISCUSSION TITLE
Integrating YARP with Consul

### DISCUSSION THREAD

**Discussion by oshabani** [dotnet/yarp#2343] [Q&A] [answered]
Hi,

we are using YARP as a gateway API for our architecture, we need to integrate is with Consul, is there any way to do that ?
we are trying to use Consul as Service Discovery and to do the health checks 
regards,

**Comment by ReubenBond** [comment, marked as answer]
Hi @oshabani, this should be possible today using the (preview) `Microsoft.Extensions.ServiceDiscovery.Dns` and the `Microsoft.Extensions.ServiceDiscovery.Yarp` library The first library implements service discovery via DNS SRV, which [Consul supports](https://developer.hashicorp.com/consul/docs/services/discovery/dns-overview), and the second integrates service discovery into YARP using a [destination resolver](https://microsoft.github.io/reverse-proxy/articles/destination-resolvers.html). I have not tried integrating with Consul yet, so your experience would be welcome.

The README files contain some example code to get you started:
* https://github.com/dotnet/aspire/tree/main/src/Microsoft.Extensions.ServiceDiscovery.Yarp
* https://github.com/dotnet/aspire/tree/main/src/Microsoft.Extensions.ServiceDiscovery.Dns

I hope that helps. Please let me know if I can provide more information.
